src/main/java/us/kbase/mobu/ModuleBuilder.java

```diff
@@ -27,669 +27,704 @@
-
-	private static int runInstallCommand(InstallCommandArgs installArgs, JCommander jc) {
-		if (installArgs.moduleName == null || installArgs.moduleName.size() != 1) {
-			ModuleBuilder.showError("Command Line Argument Error", 
-					"One and only one module name should be provided");
-			return 1;
-		}
-		try {
-			return new ClientInstaller().install(installArgs.lang, installArgs.async,
-					installArgs.core || installArgs.sync, installArgs.dynamic, 
-					installArgs.tagVer, installArgs.verbose, installArgs.moduleName.get(0), 
-					null, installArgs.clientName);
-		} catch (Exception e) {
-			if (installArgs.verbose)
-				e.printStackTrace();
-			showError("Error while installing client", e.getMessage());
-			return 1;
-		}
+	
+	public static class TagAndVerbose extends Verbose {
+		
+		@Option(
+				paramLabel = "<tag_or_version>",
+				names = {"-t", "--tag-or-ver"},
+				description = """
+						Set the version of the module to run on a call - \
+						one of the "dev", "beta", or "release" tags or a git hash. \
+						The targeted version must be registered in the KBase Catalog. \
+						The default is based on the latest release in the KBase Catalog.\
+						"""
+		)
+		String tagVer;
 	}
-
-	private static int runRunCommand(RunCommandArgs runArgs, JCommander jc) {
-		if (runArgs.methodName == null || runArgs.methodName.size() != 1) {
-			ModuleBuilder.showError("Command Line Argument Error", 
-					"One and only one method name should be provided");
-			return 1;
-		}
-		try {
-			if (runArgs.inputFile == null && runArgs.inputJson == null && !runArgs.stdin)
-				throw new IllegalStateException("No one input method is used " +
-						"(should be one of '-i', '-j' or '-s')");
-			return new ModuleRunner(runArgs.sdkHome).run(runArgs.methodName.get(0), 
-					runArgs.inputFile, runArgs.stdin, runArgs.inputJson, runArgs.output, 
-					runArgs.tagVer, runArgs.verbose, runArgs.keepTempFiles, runArgs.provRefs, 
-					runArgs.mountPoints);
-		} catch (Exception e) {
-			if (runArgs.verbose)
-				e.printStackTrace();
-			showError("Error while installing client", e.getMessage());
-			return 1;
+	
+	@Command(name = "install", description = "Install a client for a KBase module.")
+	public static class InstallCommand extends TagAndVerbose implements Callable<Integer> {
+		
+		@Option(
+				names = {"-l", "--language"},
+				description = """
+						Language for the generated client. Defaults to the module langage. \
+						Valid values: ${COMPLETION-CANDIDATES}.\
+						"""
+		)
+		Language language;
+		
+		@Option(
+				names = {"-a", "--async"},
+				description = """
+						Force generation of asynchronous calls. Default is the KBase Catalog \
+						setting for the module.\
+						""",
+				defaultValue = "false"
+		)
+		boolean async;
+		
+		@Option(
+				names = {"-c", "--core"},
+				description = """
+						Force generation of synchronous calls for KBase core services. \
+						In almost all cases specifying this argument is a mistake.\
```

Comment:
haha!

Reply:
yeah, I can't really think of a use case where specifying this option makes sense

Reply:
I wonder if there are any services that use that option.

Reply:
I don't think so - basically either you're either providing a module name, in which case it'll be either compiled as async or dynserv depending on the catalog, or a spec, in which case it'll be compiled as sync. That seems like it should work for all situations I can think of. Maybe I'm missing something though, which is why I left it in